src/coreclr/tools/Common/TypeSystem/Common/ExplicitLayoutValidator.cs

```diff
@@ -125,27 +179,133 @@ private void MarkORefLocations(MetadataType type, bool[] orefMap, int offset)
             }
         }
 
-        private void SetFieldLayout(int offset, int count, FieldLayoutTag tag)
+        private void SetFieldLayout(List<FieldLayoutInterval> fieldLayoutInterval, int offset, int count, FieldLayoutTag tag)
         {
-            for (int index = 0; index < count; index++)
+            if (count == 0)
+                return;
+
+            var newInterval = new FieldLayoutInterval(offset, count, tag);
+
+            int binarySearchIndex = fieldLayoutInterval.BinarySearch(new FieldLayoutInterval(offset, 0, FieldLayoutTag.Empty));
```

Comment:
Nit - why can't we just <code>BinarySearch(newInterval)</code>?

Reply:
We can, I just forgot to do that when I refactored the logic to make that the obvious thing to do.